domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/Transfer.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020. MobilityData IO.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.domain.entity;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class Transfer {
+
+    @NotNull
+    private final String fromStopId;
+
+    @NotNull
+    private final String toStopId;
+
+    @NotNull
+    private final TransferType transferType;
+
+    private final int minTransferTime;
```

Comment:
Should this be `@Nullable`?

Reply:
From what I witnessed, IntelliJ recommends to not annotate `@Nullable` primitives such as int. I had a similar discussion about object types with @fabrice-v .

Reply:
Indeed since it's optional, this should be annotated as `@Nullable`. Therefore, we'll use object type Integer instead of primitive int to allow use of such annotation.

Reply:
Oh, right, I missed that this was an `int`. But then how are you checking if you should validate this field? If other words, later in the pipeline, how do you know if you should check the optional field with some kind of rule?

Reply:
Ok, seeing your edited response now. Yes, if you switch to the object representation that works.